pkg/controller/shoot/shoot_control_delete.go

```diff
@@ -118,10 +118,10 @@ func (c *defaultControl) deleteShoot(o *operation.Operation) *gardenv1beta1.Last
 		destroyExternalDomainDNSRecord = f.AddTask(botanist.DestroyExternalDomainDNSRecord, 0, cleanKubernetesResources)
 		syncPointTerraformers          = f.AddSyncPoint(deleteSeedMonitoring, destroyNginxIngressResources, destroyKube2IAMResources, destroyInfrastructure, destroyExternalDomainDNSRecord)
 		deleteKubeAPIServer            = f.AddTask(botanist.DeleteKubeAPIServer, defaultRetry, syncPointTerraformers)
-		destroyBackupInfrastructure    = f.AddTask(seedCloudBotanist.DestroyBackupInfrastructure, 0, deleteKubeAPIServer)
+		deleteBackupInfrastructure     = f.AddTask(botanist.DeleteBackupInfrastructure, 0, deleteKubeAPIServer)
```

Comment:
Backwards compatibility not handled in deletion case.
What happens when an old Shoot is already marked as to-be-deleted? No BackupInfrastructure resource will be created, but the backup Terraform state will still be in the regular shoot namespace in the seed.

Reply:
Hmm. I didn't handle that case. Will probably  have a check on `deletionTimestamp` and presence of TF resources main seedNamespace. By to-de-deleted you mean shoot set with ConfirmationDeletionTimestamp annotation prior to upgrade right?

Reply:
I think the easiest solution is to also put the migration parts into the deletion flow at the beginning (move Terraform resources and create BackupInfrastructure object). Then do the deletion regularly without special handling.

Reply:
Yeah we can use that. Thanks.